src/libraries/Microsoft.Extensions.Diagnostics/src/Metrics/ListenerSubscription.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+#if NET
+using System.Runtime.CompilerServices;
+#endif
+
+namespace Microsoft.Extensions.Diagnostics.Metrics
+{
+    internal class ListenerSubscription : IObservableInstrumentsSource, IDisposable
+    {
+        private readonly MeterListener _meterListener;
+        private readonly IMetricsListener _metricsListener;
+#if NET
+        private readonly ConditionalWeakTable<Instrument, InstrumentStatus> _instruments = new();
```

Comment:
The main reason to track all of the instruments was so that we could enable them later if the rules changed. Using a second MeterListener to enumerate them is an interesting trick, I'll give it a try.